core/sdk/src/api/manifest/device_manifest.rs

```diff
@@ -68,6 +68,8 @@ pub struct RippleConfiguration {
     pub metrics_logging_percentage: u32,
     #[serde(default)]
     pub internet_monitoring_configuration: InternetMonitoringConfiguration,
+    #[serde(default = "log_signal_default_level")]
+    pub log_signal_log_level: String,
```

Comment:
I think if you make this a `log::LevelFilter` instead of a `String` then you don't have to do the to_str()/unwrap_or logic later.